pandas/core/arrays/base.py

```diff
@@ -477,6 +486,16 @@ def take(self, indexer, allow_fill=True, fill_value=None):
            def take(self, indexer, allow_fill=True, fill_value=None):
                indexer = np.asarray(indexer)
                mask = indexer == -1
+
+               # take on empty array not handled as desired by numpy
+               if not len(self):
+                   # only valid if result is an all-missing array
+                   if mask.all():
+                       return type(self)([self._na_value] * len(indexer))
+                   else:
```

Comment:
you don't need the else, just raise

Reply:
Yeah, the 'else' clause is indeed not essential. I adapted the error message slightly ("from empty axes" -> "from empty array") because I did not find it the best for ExtensionArrays, but given the minor change I can certainly leave it out for code simplicity.

Reply:
pls make this change.

also combine the comments, this doesn't read very well

Reply:
yep, did it already locally, didn't push yet